src/audio/wiiu/SDL_wiiuaudio.c

```diff
@@ -249,9 +281,43 @@ static void _WIIUAUDIO_ThreadDeallocator(OSThread *thread, void *stack) {
 static void _WIIUAUDIO_ThreadCleanup(OSThread *thread, void *stack) {
 }
 
+static void WIIUAUDIO_DetectDevices(void) {
+    void *drcHandle;
+    void *tvHandle; 
+    void *mirrorHandle;
+
+    /* This gets reset later anyways */
+    SDL_AudioSpec spec;
+
+    spec.channels = WIIU_MAX_VALID_CHANNELS;
+    spec.format = AUDIO_S16MSB;
+    spec.samples = 4096;
+
+    SDL_CalculateAudioSpec(&spec);
+
+    SDL_AddAudioDevice(SDL_FALSE, SDL_AUDIO_DEVICE_WIIU_MIRRORED, &spec, &mirrorHandle);
+    SDL_AddAudioDevice(SDL_FALSE, SDL_AUDIO_DEVICE_WIIU_TV, &spec, &tvHandle);
+    SDL_AddAudioDevice(SDL_FALSE, SDL_AUDIO_DEVICE_WIIU_GAMEPAD, &spec, &drcHandle);
```

Comment:
The "handles" here are supposed to be unique identifiers/pointers which can be retried using `device->handle`. In this case you could do something similar to the SNDIO driver, and use your device IDs here:
https://github.com/devkitPro/SDL/blob/8289656a4ec11600045e411177680f974991953e/src/audio/sndio/SDL_sndioaudio.c#L324-L328
Then you can check this handle in `WIIUAUDIO_OpenDevice` and avoid comparing against the device name.